### PR TITLE
Modal: slight JS cleanup, add email validation

### PR DESCRIPTION
**Pull request by Rio517**
This prevents us from getting empty email submission on subsequent
nominations

### CODE REVIEW

File: _includes/panel_action_modal.html [hunk 1]
```diff
@@ -106,46 +105,39 @@
...
-      $("button.selectpicker").focus();
-    } else {
-      var saveURI = '{{site.services_url}}/nominations';
-      $.post(saveURI, $(this).serialize());
-      $("#nomination-page-1").hide();
-      $("#nomination-page-2").show();
-      $("#additional-fields").hide();
-      $("textarea[name=selection_comment]").val("");
-      $("textarea[name=other_comment]").val("");
-      $("select[name=legislator_id]").val("")
```
Comment (turino) [inline _includes/panel_action_modal.html:146]: @Rio517 is there a reason you don't want to reset these form fields?

Comment (Rio517) [inline _includes/panel_action_modal.html:146]: Yes. This prevents us from getting empty email submission on subsequent
nominations. I'm resetting some fields on line 119, but don't want to reset the whole form.

Comment (turino) [inline _includes/panel_action_modal.html:146]: The fields that were being cleared before were the same as you're now clearing on line 119, but I think you'll want to be sure they get cleared before `submit-again`, which is no longer happening

Comment (turino) [inline _includes/panel_action_modal.html:146]: note I'm just referring to the change in `$("#js-nomination-form").submit()`, not in the `show-nomination-modal` function

Comment (Rio517) [inline _includes/panel_action_modal.html:146]: It does happen.  It happens before the form gets displayed again - right before the show is actually called.  I moved it to better associate the code to when it's actually required to be called.  This way if someone is quickly glacing they won't wonder why were clearing the form, but clearly see the related call to unhide it.

However, It actually seems like .val("") and what I wrote above are functionally equivalent.  I should just do `$("textarea, select").val("")`.  Will update later.

Comment (turino) [inline _includes/panel_action_modal.html:146]: right, but the `submit-again` button (line 128) doesn't call that code

**Pull request merged**